src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -69,23 +113,19 @@ - (void)runLoop
         unsigned int delaySeconds = (10 * 60); // 10 minutes.
         DDLogDebug(
             @"%@ No more expiring messages. Setting next check %u seconds into the future", self.tag, delaySeconds);
-        [self runBy:now + delaySeconds * 1000];
+        [self runByDate:[NSDate ows_dateWithMillisecondsSince1970:now + delaySeconds * 1000]];
```

Comment:
Is this `if` block still necessary? It seems redundant with intent of kMaxDelaySeconds.